.github/workflows/release-wasi-demo-app.yml

```diff
@@ -0,0 +1,139 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-action.json
+
+name: Release wasi-demo-app
+run-name: wasi-demo-app@${{ inputs.version }}
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "The version of the image to release. (e.g., 1.2.3)"
+        type: string
+        required: true
+      dry_run:
+        description: "Run the release without actually releasing bits"
+        type: boolean
+        default: true
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  release-wasi-demo:
+    permissions:
+      packages: write
+    runs-on: ubuntu-latest
+    outputs:
+      wasi_demo_app_digest: ${{ steps.get_digests.outputs.wasi_demo_app_digest }}
+      wasi_demo_oci_digest: ${{ steps.get_digests.outputs.wasi_demo_oci_digest }}
+      wasi_demo_oci_artifact_digest: ${{ steps.get_digests.outputs.wasi_demo_oci_artifact_digest }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup build env
+        run: ./scripts/setup-linux.sh
+      - name: Install Rust and wasm32-wasi target
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          target: wasm32-wasip1
+          override: true
+      - name: Build oci tarballs
+        run: |
+          make load
+          make load/oci
+          sudo ctr image ls
+      - name: Tagging images
+        run: |
+          sudo ctr image tag ghcr.io/containerd/runwasi/wasi-demo-app:latest ghcr.io/${{ github.repository }}/wasi-demo-app:${{ inputs.version }}
+          sudo ctr image tag ghcr.io/containerd/runwasi/wasi-demo-oci:latest ghcr.io/${{ github.repository }}/wasi-demo-app:latest
+
+          sudo ctr image tag ghcr.io/containerd/runwasi/wasi-demo-oci:latest ghcr.io/${{ github.repository }}/wasi-demo-oci:${{ inputs.version }}
+          sudo ctr image tag ghcr.io/containerd/runwasi/wasi-demo-oci:latest ghcr.io/${{ github.repository }}/wasi-demo-oci:latest
+
+          sudo ctr image tag ghcr.io/containerd/runwasi/wasi-demo-oci-artifact:latest ghcr.io/${{ github.repository }}/wasi-demo-oci-artifact:${{ inputs.version }}
+          sudo ctr image tag ghcr.io/containerd/runwasi/wasi-demo-oci-artifact:latest ghcr.io/${{ github.repository }}/wasi-demo-oci-artifact:latest
+      
+      - name: Push images to ghcr.io
+        if: ${{ inputs.dry_run == false }}
+        run: |
+          sudo ctr images push --user ${{ github.actor }}:${{ secrets.GITHUB_TOKEN }} ghcr.io/${{ github.repository }}/wasi-demo-app:${{ inputs.version }}
```

Comment:
is this going to leak our secret?

Reply:
This is what it looks like on the CI logs
![image](https://github.com/user-attachments/assets/a4c8d908-517e-4570-bca1-9fa2f540e727)

I believe it won't leak the secrets.

Reply:
I pushed a commit that uses docker instead of `ctr`